atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionOutcomeMetricsAssert.java

```diff
@@ -0,0 +1,140 @@
+        return new TransactionOutcomeMetricsAssert(actual);
+    }
+
+    public TransactionOutcomeMetricsAssert hasSuccessfulCommits(long count) {
+        checkPresentAndCheckCount("successfulCommit", count);
+        return this;
+    }
+
+    public TransactionOutcomeMetricsAssert hasFailedCommits(long count) {
+        checkPresentAndCheckCount("failedCommit", count);
+        return this;
+    }
+
+    public TransactionOutcomeMetricsAssert hasAborts(long count) {
+        checkPresentAndCheckCount("abort", count);
+        return this;
+    }
+
+    public TransactionOutcomeMetricsAssert hasRollbackOther(long count) {
+        checkPresentAndCheckCount("rollbackOther", count);
+        return this;
+    }
+
+    public TransactionOutcomeMetricsAssert hasLocksExpired(long count) {
+        checkPresentAndCheckCount("locksExpired", count);
+        return this;
+    }
+
+    public TransactionOutcomeMetricsAssert hasPutUnlessExistsFailed(long count) {
+        checkPresentAndCheckCount("putUnlessExistsFailed", count);
+        return this;
+    }
+
+    public TransactionOutcomeMetricsAssert hasNamedReadWriteConflicts(TableReference tableReference, long count) {
+        checkPresentAndCheckCount("readWriteConflict", count, tableReference);
+        return this;
+    }
+
+    public TransactionOutcomeMetricsAssert hasNamedWriteWriteConflicts(TableReference tableReference, long count) {
+        checkPresentAndCheckCount("writeWriteConflict", count, tableReference);
+        return this;
+    }
+
+    public TransactionOutcomeMetricsAssert hasPlaceholderReadWriteConflicts(long count) {
+        checkPresentAndCheckCount("readWriteConflict", count, LoggingArgs.PLACEHOLDER_TABLE_REFERENCE);
+        return this;
+    }
+
+    public TransactionOutcomeMetricsAssert hasPlaceholderWriteWriteConflicts(long count) {
+        checkPresentAndCheckCount("writeWriteConflict", count, LoggingArgs.PLACEHOLDER_TABLE_REFERENCE);
+        return this;
+    }
+
+    public TransactionOutcomeMetricsAssert hasPlaceholderWriteWriteConflictsSatisfying(LongConsumer assertion) {
+        MetricName metricName = actual.getMetricName("writeWriteConflict",
+                getTableReferenceTags(LoggingArgs.PLACEHOLDER_TABLE_REFERENCE));
+        assertion.accept(taggedMetricRegistry.meter(metricName).getCount());
+        return this;
+    }
+
+    public TransactionOutcomeMetricsAssert hasNoKnowledgeOf(TableReference tableReference) {
+        assertMetricNotExists(actual.getMetricName("readWriteConflict", getTableReferenceTags(tableReference)));
+        assertMetricNotExists(actual.getMetricName("writeWriteConflict", getTableReferenceTags(tableReference)));
+        return this;
+    }
+
+    private ImmutableMap<String, String> getTableReferenceTags(TableReference tableReference) {
+        return ImmutableMap.of("tableReference", tableReference.getQualifiedName());
+    }
+
+    private void checkPresentAndCheckCount(String name, long count) {
```

Comment:
strengthening argument for the optional change above:
1 - `(read|write)WriteConflict` appears 3-4 times per operation
2 - these `String` parameters can become `TransactionOutcomeMetric` parameters

Reply:
Done as part of the Enum change.